lib/bundler/fetcher/downloader.rb

```diff
@@ -36,6 +36,7 @@ def fetch(uri, counter = 0)
 
       def request(uri)
         Bundler.ui.debug "HTTP GET #{uri}"
+        raise GemfileError, "Incorrect gem source URL: #{uri}" unless uri.request_uri
```

Comment:
This shouldn't be raising a `GemfileError`, and are you 100% certain that `uri.request_uri` will return nil instead of throwing?